### PR TITLE
Nitzan/eco 758/fix webview mailto

### PR DESCRIPTION
**Pull request by nitzantomer**
Our webview did not support `mailto` links.
This PR fixes that.

### CODE REVIEW

**Comment by phips28** [comment]
pls also merge it into `no_loc` branch when its ready for Swelly release. Thx!

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebViewClient.java [hunk 1]
```diff
@@ -24,4 +31,29 @@ public void onPageFinished(WebView view, String url) {
     public void onReceivedError(WebView view, WebResourceRequest request, WebResourceError error) {
 		Logger.log(new Log().withTag(TAG).text("onReceivedError(WEB_VIEW, REQUEST, \"" + error.toString() + "\")"));
     }
+
+    @Override
+    public boolean shouldOverrideUrlLoading(WebView view, String url) {
```
Comment (YohayBar) [inline sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebViewClient.java:36]: From documentation - 
This method was deprecated in API level 24.
Use shouldOverrideUrlLoading(WebView, WebResourceRequest) instead.

**Review by YohayBar — APPROVED** [review]
👍 fix conflicts

File: sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebViewClient.java [hunk 2]
```diff
@@ -24,4 +31,29 @@ public void onPageFinished(WebView view, String url) {
...
+        if (url.startsWith("mailto:")) {
+            MailTo mt = MailTo.parse(url);
+            Intent intent = newEmailIntent(context, mt.getTo(), mt.getSubject(), mt.getBody(), mt.getCc());
+            context.startActivity(intent);
+
+            return true;
+        } else {
+            view.loadUrl(url);
```
Comment (YohayBar) [inline sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebViewClient.java]: ?

Comment (YohayBar) [reply sdk/src/main/java/com/kin/ecosystem/web/EcosystemWebViewClient.java]: "Note: Do not call WebView.loadUrl(String) with the same URL and then return true. This unnecessarily cancels the current load and starts a new load with the same URL. The correct way to continue loading a given URL is to simply return false, without calling WebView.loadUrl(String)."

**Review by YohayBar — APPROVED** [review]

**Pull request merged**